### PR TITLE
Add Themes Support

### PR DESCRIPTION
**Pull request by ReenigneArcher**
## Description

Please include a summary of the change and which issue is fixed.
- Closely replicates Poster Mixins for Themes
  - list themes
  - upload theme
  - set theme
  - get theme url with token
- Add Theme attribute for movies in video.py
- Fixes typo in PosterUrlMixin class (`grandparentThumb`)
- Replaces #658 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [ ] I have added tests when applicable

### CODE REVIEW

**Comment by ReenigneArcher** [comment]
Again, I understand if you don't want to merge this due to not being able to easily delete the Themes. Either way, this is a cleaner version of #658 .

**Comment by JonnyWong16** [comment]
Can you separate this into a new bugfix PR?

* Fixes typo in PosterUrlMixin class (`grandparentThumb`)

File: plexapi/media.py [hunk 1]
```diff
@@ -1106,3 +1106,29 @@ def _loadData(self, data):
     @deprecated('use "languageCodes" instead')
     def languageCode(self):
         return self.languageCodes
+
+
+class Theme(PlexObject):
```
Comment (JonnyWong16) [inline plexapi/media.py:1111]: Move up after `Poster`.

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]
`Collection` can also have `ThemeMixin`.

File: plexapi/media.py (hunk 1 above)
Comment (JonnyWong16) [reply plexapi/media.py:1111]: Actually just subclass `BaseImage` with a different `TAG`.

Comment (ReenigneArcher) [reply plexapi/media.py:1111]: > Actually just subclass `BaseImage` with a different `TAG`.

There is no `provider` or `thumb` for this. Is it still preferable to subclass it as `BaseImage`?

Comment (JonnyWong16) [reply plexapi/media.py:1111]: There actually is a `thumb`.

Yes, subclass. `provider` will just be `None`.

**Comment by ReenigneArcher** [comment]
> Can you separate this into a new bugfix PR?
> 
>     * Fixes typo in PosterUrlMixin class (`grandparentThumb`)

Do I also need to revert that change in this branch?

**Comment by JonnyWong16** [comment]
> Do I also need to revert that change in this branch?

No, leave it for now and rebase after #880.

File: plexapi/media.py [hunk 2]
```diff
@@ -917,18 +917,18 @@ def _loadData(self, data):
...
-            ratingKey (str): Unique key identifying the poster or art.
-            selected (bool): True if the poster or art is currently selected.
-            thumb (str): The URL to retrieve the poster or art thumbnail.
+            provider (str): The source of the art or poster, None for Theme objects.
+            ratingKey (str): Unique key identifying the resource.
+            selected (bool): True if the resource is currently selected.
+            thumb (str): The URL to retrieve the resource thumbnail.
     """
-    TAG = 'Photo'
+    TAG = None
```
Comment (JonnyWong16) [inline plexapi/media.py:931]: `PlexObject` already sets `TAG = None`. Don't need it here.

File: plexapi/media.py [hunk 3]
```diff
@@ -947,16 +947,33 @@ def select(self):
...
-class Poster(BaseImage):
+class Poster(BaseResource):
     """ Represents a single Poster object. """
+    TAG = 'Photo'
+
+
+class Theme(BaseResource):
+    """ Represents a single Theme object.
+
+        Attributes:
```
Comment (JonnyWong16) [inline plexapi/media.py:968]: Don't need doc string here.

**Review by JonnyWong16 — CHANGES_REQUESTED** [review]

**Review by JonnyWong16 — APPROVED** [review]

**Pull request merged**